kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/StatsSchemaHelper.java

```diff
@@ -56,27 +56,16 @@ public class StatsSchemaHelper {
    * limited set of data types and only literals of those types are skipping eligible.
    */
   public static boolean isSkippingEligibleLiteral(Literal literal) {
-    // Even if the literal's data type is not `StringType` and it is used in a collation-aware
-    // `Predicate`, it can be cast to `StringType` and used for collation-aware skipping.
-    // Therefore, we check for non-collated skipping eligibility here.
-    // For columns, we can't do this because we don't have statistics for non-`StringType` columns
-    // under collation-aware stats.
-    return isSkippingEligibleDataType(literal.getDataType(), false);
+    return isSkippingEligibleDataType(literal.getDataType());
   }
 
   /** Returns true if the given data type is eligible for MIN/MAX data skipping. */
-  public static boolean isSkippingEligibleDataType(DataType dataType, boolean isCollatedSkipping) {
```

Comment:
So the main reason for doing this was to avoid expressions that have a collation but don't operate on a string column? what will happen now instead? it will propogate down to the engine to fail?

Reply:
makes sense, we weren't doing this check for the literal anyways it seems